packages/ethereum/contracts/HoprChannels/Channels.sol

```diff
@@ -0,0 +1,283 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract Channels {
+    /**
+     * @dev Possible channel statuses.
+     * We find out the channel's status by
+     * using {_getChannelStatus}.
+     */
+    enum ChannelStatus { CLOSED, OPEN, PENDING_TO_CLOSE }
+
+    /**
+     * @dev A channel struct
+     */
+    struct Channel {
```

Comment:
@nionis what are the gas implications of this? Looks hugely wasteful compared to what we had before here. I guess that makes e.g. channel opening and redemption way more expensive. IMO the byte shifting was not a huge issue and is done in various contracts in production as it leads to massively fewer words being written into storage.

Reply:
PR is not ready for review

Reply:
(also this is not different from what we had before)